tests/modules/site-health/audit-full-page-cache/audit-full-page-cache-test.php

```diff
@@ -0,0 +1,367 @@
+					return $threshold;
+				}
+			);
+		}
+
+		add_filter(
+			'pre_http_request',
+			function ( $r, $parsed_args ) use ( &$responses, &$is_unauthorized, $good_basic_auth, $delay_the_response, $threshold ) {
+
+				$expected_response = array_shift( $responses );
+
+				if ( $delay_the_response ) {
+					usleep( $threshold * 1000 + 1 );
+				}
+
+				if ( 'unauthorized' === $expected_response ) {
+					$is_unauthorized = true;
+
+					return array(
+						'response' => array(
+							'code'    => 401,
+							'message' => 'Unauthorized',
+						),
+					);
+				}
+
+				if ( null !== $good_basic_auth ) {
+					$this->assertArrayHasKey(
+						'Authorization',
+						$parsed_args['headers']
+					);
+				}
+
+				$this->assertIsArray( $expected_response );
+
+				return array(
+					'headers'  => $expected_response,
+					'response' => array(
+						'code'    => 200,
+						'message' => 'OK',
+					),
+				);
+			},
+			20,
+			2
+		);
+
+		$actual = perflab_afpc_page_cache_test();
+		$this->assertArrayHasKey( 'description', $actual );
+		$this->assertArrayHasKey( 'actions', $actual );
+		if ( $is_unauthorized ) {
+			$this->assertStringContainsString( 'Unauthorized', $actual['description'] );
+		} else {
+			$this->assertStringNotContainsString( 'Unauthorized', $actual['description'] );
+		}
+
+		$this->assertEquals(
+			$expected_props,
+			wp_array_slice_assoc( $actual, array_keys( $expected_props ) )
+		);
+	}
+
+	/**
+	 * Tests Rest endpoint registration.
+	 */
+	public function test_perflab_afpc_register_async_test_endpoints() {
+		$server = rest_get_server();
+		$routes = $server->get_routes();
+
+		$endpoint = '/' . self::REST_NAMESPACE . self::REST_ROTE;
+		$this->assertArrayHasKey( $endpoint, $routes );
+
+		$route = $routes[ $endpoint ];
+		$this->assertCount( 1, $route );
+
+		$route = current( $route );
+		$this->assertEquals(
+			array( WP_REST_Server::READABLE => true ),
+			$route['methods']
+		);
+
+		$this->assertEquals(
+			'perflab_afpc_page_cache_test',
+			$route['callback']
+		);
+
+		$this->assertIsCallable( $route['permission_callback'] );
+
+		if ( current_user_can( 'view_site_health_checks' ) ) {
+			$this->assertTrue( call_user_func( $route['permission_callback'] ) );
+		} else {
+			$this->assertFalse( call_user_func( $route['permission_callback'] ) );
+		}
+
+		wp_set_current_user( self::factory()->user->create( array( 'role' => 'author' ) ) );
+		$this->assertFalse( call_user_func( $route['permission_callback'] ) );
+
+		$user = wp_set_current_user( self::factory()->user->create( array( 'role' => 'administrator' ) ) );
+		if ( is_multisite() ) {
+			// Site health cap is only available for super admins in Multi sites.
+			grant_super_admin( $user->ID );
+		}
+		$this->assertTrue( call_user_func( $route['permission_callback'] ) );
+	}
+
+	/**
+	 * Gets response data for test_perflab_afpc_page_cache_test().
+	 *
+	 * @return array[]
+	 */
+	public function provider_perflab_afpc_page_cache_test() {
```

Comment:
A bit of a nit-pick, but could you move this provider method right below the test that it belongs to? That's how core structures it, and it helps for a better code overview when test and provider are colocated.